src/test/java/com/google/api/codegen/testdata/go_library.baseline

```diff
@@ -462,166 +462,153 @@ func (client *Client) ListStrings(ctx context.Context, req *google_example_libra
     var callOpts []gax.CallOption
     callOpts = append(callOpts, client.callOptions["ListStrings"]...)
     callOpts = append(callOpts, opts...)
-    atLastPage := false
-    return &StringIterator{
-        apiCall: func() (StringPage, error) {
-            if atLastPage {
-                return StringPage{}, io.EOF
-            }
-            var resp *google_example_library_v1.ListStringsResponse
-            err := gax.Invoke(ctx, func (c context.Context) error {
-                var err error
-                resp, err = client.client.ListStrings(c, req)
-                return err
-            }, callOpts...)
-            if err != nil {
-                return StringPage{}, err
-            }
-            if resp.NextPageToken == "" {
-                atLastPage = true
-            } else {
-                req.PageToken = resp.NextPageToken
-            }
-            return StringPage{
-                Items: resp.Strings,
-                NextPageToken: resp.NextPageToken,
-            }, nil
-        },
-    }, nil
+
+    iterator := &StringIterator{}
+    iterator.apiCall = func() error {
+        if iterator.atLastPage {
+            return Done
+        }
+        var resp *google_example_library_v1.ListStringsResponse
+        err := gax.Invoke(ctx, func (c context.Context) error {
+            var err error
+            req.PageToken = iterator.nextPageToken
```

Comment:
Yes, or very nearly. I'm about to change the recommendation to:


   - Next() and NextPage() on the same iterator.
   - Iterator has methods:
   - SetPageSize(int32)
      - SetPageToken(string)
      - NextPageToken() string


On Fri, May 20, 2016 at 12:25 AM Sai Cheemalapati <notifications@github.com>
wrote:

> In src/test/java/com/google/api/codegen/testdata/go_library.baseline
> <https://github.com/googleapis/toolkit/pull/123#discussion_r63989765>:
>
> > -            return StringPage{
> > -                Items: resp.Strings,
> > -                NextPageToken: resp.NextPageToken,
> > -            }, nil
> > -        },
> > -    }, nil
> > +
> > +    iterator := &StringIterator{}
> > +    iterator.apiCall = func() error {
> > +        if iterator.atLastPage {
> > +            return Done
> > +        }
> > +        var resp *google_example_library_v1.ListStringsResponse
> > +        err := gax.Invoke(ctx, func (c context.Context) error {
> > +            var err error
> > +            req.PageToken = iterator.nextPageToken
>
> Think the discussion is settled?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/googleapis/toolkit/pull/123/files/739520f74c66bf422738b1c4242b7ff7540a6342#r63989765>
>